### ISSUE TITLE
Make Level Drawings and Animation Columns independent of each other

### ISSUE THREAD

**Issue by amadlover** [opentoonz/opentoonz#4297]
**Question**

Deleting a column in the animation timeline deletes the level drawings from the scene as well. There is always a need to keep a Level in the scene without having in the animation column. Currently there has to a column in the animation timeline referencing atleast 1 drawing from a particular level for the level drawings to be kept in the scene.

Should there not be a workflow where there can be x number of levels and y number of columns both independent of each other.

Using the 1.5 version of OpenToonz through the binary installer.

Best regards
Nihal Kenkre.

**Comment by artisteacher** [comment]
Deleting a column doesn't delete the level drawings. It only removes their exposure from the scene. If you look in the scene cast, they should still be associated with the scene - you can add them back at any time. In an empty cell, just type the level name, space, then the drawing number (ie. B 1) and the drawing can be brought back. Or right-click and load the level. Or right-click on the level in the scene cast and "Expose in Xsheet".

You can already have any number of levels in a column - so the number of levels vs. columns doesn't have to correspond.

**Comment by gab3d** [comment]
On the subject:
There's a new option in the Preferences (that will come in v1.6) allowing to effectively delete the levels from the scene, when they're not being used anywhere.

![imagen](https://user-images.githubusercontent.com/18104091/156832523-c2388de4-a6c7-4edb-90e0-0fdcf08dcbc3.png)

**Comment by RodneyBaker** [comment]
@artisteacher Thanks for that reminder.  That is yet another reason why users should consider using Scene Cast window.
It is perhaps too easy to think a level is gone forever when it:  1) disappears from xsheet/timeline and level strip 2) cannot be found in the drawings directory.  The average user will not know to look to Scene Cast to recover the level.

This of course especially true regarding levels that have never been saved.

I do think something could probably be done to smooth out the user experience when removing the last exposure of a level from the xsheet/timeline.  At least to the point where it is more obvious the level has not been completely lost.

At any rate, I appreciate the reminder because I've often found myself thinking a level is entirely gone when it's actually still in memory and available via Screencast.  Note to self!

**Comment by gab3d** [comment]
One thing I've proposed in "some other issue", is that when **Expose Loaded Levels in Xsheet** option is turned off in Preferences, the program automatically opens up a window with the Scene Cast panel (if it's not already present in the current room), so users would know where the h*ll the loaded level is.

I suppose some sort of soft warning could also be issued when deleting the last ocurrence of a level in the scene, to tell users to check Scene Cast in case that level is needed in the future...

**Comment by amadlover** [comment]
![level_column_1](https://user-images.githubusercontent.com/463752/156984208-2a558b1b-f94e-4411-8fbd-73848eee1182.png)

Above is the Level Strip Ex06 exposed in Column6 of the XSheet.

![level_column_2](https://user-images.githubusercontent.com/463752/156984399-c54305ee-2f84-4628-b8f7-b21bded9a139.png)

Above is what happens when Colume6 is deleted from the XSheet. Since there is no reference to the drawings of the level strip any where in the XSheet the Level Strip for Ex06 is removed.

Would it be better to have the Level Strips present independant of the XSheet exposure. Much like somebody creating a bunch of drawings and then choosing to expose as and when and where they are required in the XSheet.

**Comment by RodneyBaker** [comment]
@amadlover 

I'm a little confused here.  You say:
Would it be better to have the Level Strips present independant of the XSheet exposure. Much like somebody creating a bunch of drawings and then choosing to expose as and when and where they are required in the XSheet.

That's how the Level Strip/level and Xsheet-Timeline/exposures work now.

As @artisteacher and @gab3d suggest...

Try removing a column and/or all levels exposed while the Scene Cast window is open.
You'll see the levels there before they are removed and they'll still be there after removal.
We can drag them back into the xsheet from the Scene Cast windows or Right Click and use an appropriate optio to either expose the level in Xsheet/Timeline or open the level in the Level Editor:

![image](https://user-images.githubusercontent.com/2576432/181688599-ebeb027c-c909-42e6-afdf-19f6ef86eddd.png)

**Comment by RodneyBaker** [comment]
I do think it would be useful to have the Level Strip list the levels that are still present but not active in the Level Strip's dropdown menu.  They the user could pull that level back in that way.

Currently, even though other unused levels are present (as shown in previous post) they aren't in the dropdown menu:
![image](https://user-images.githubusercontent.com/2576432/181688969-124185f0-79cd-4390-acd9-e083dc37bd4c.png)

**Comment by amadlover** [comment]
Hello.

Thank you for letting me know the workflows around the level strips and xsheet columns.  I did not know about the Scene cast window at all.

> Currently, even though other unused levels are present (as shown in previous post) they aren't in the dropdown menu: ![image](https://user-images.githubusercontent.com/2576432/181688969-124185f0-79cd-4390-acd9-e083dc37bd4c.png)

As you mentioned it would be useful to have the level strips be present in the dropdown even though they are not a part of the exposure scene. like holding the drawings in hand before shooting them. 

Closing this for now. 

Thank you,
Nihal Kenkre.
